### PR TITLE
RD-11321-otel-distros-python-and-node-filter-out-empty-sqs-polls

### PR DESCRIPTION
**Pull request by sagivoululumigo**

### CODE REVIEW

File: src/lumigo_opentelemetry/libs/general_utils.py [hunk 1]
```diff
@@ -70,3 +70,26 @@ def get_max_size() -> int:
...
+    """
+    This function return the boolean value of the given environment variable.
+    If this values doesn't exist, return default.
+
+    @param env_var_name: The env var to get (case-sensitive)
+    @param default: Default value if env var is not set
+    @return: The boolean value of the env var
+    """
+
+    env_var_value = os.environ.get(env_var_name, str(default)).lower()
```
Comment (GuyMoses) [inline src/lumigo_opentelemetry/libs/general_utils.py:85]: keep in mind that a `get` function on a dictionary in python can potentially return a `None` value,
lower() will raise an exception here in that case

File: src/lumigo_opentelemetry/resources/span_processor.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from opentelemetry.trace import Span
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+from lumigo_opentelemetry import logger
+
+
+# A span attributes that if is set to True, the span will not be exported
+SKIP_EXPORT_SPAN_ATTRIBUTE = "SKIP_EXPORT"
```
Comment (GuyMoses) [inline src/lumigo_opentelemetry/resources/span_processor.py:9]: can we import this string from sdk somehow (instead of declaring it)?

**Review by GuyMoses — CHANGES_REQUESTED** [review]

File: src/lumigo_opentelemetry/resources/span_processor.py (hunk 2 above)
Comment (sagivoululumigo) [reply src/lumigo_opentelemetry/resources/span_processor.py:9]: This isn't a standard attribute from the SDK, we made up this attribute

File: src/lumigo_opentelemetry/libs/general_utils.py (hunk 1 above)
Comment (sagivoululumigo) [reply src/lumigo_opentelemetry/libs/general_utils.py]: The `environ` is not strictly a dict, it a `class _Environ(MutableMapping)` where a value `None` should not occurs.
In any case just to be safe I added a check for if the env var value is actually `None`

File: src/lumigo_opentelemetry/resources/span_processor.py (hunk 2 above)
Comment (sagivoululumigo) [reply src/lumigo_opentelemetry/resources/span_processor.py:9]: All the standard library methods of skipping an export don't work for our case, so we are using a custom attribute.
See the discussion on the DR page (Ken responded there)
https://www.notion.so/DR-Drop-OTEL-empty-SQS-polling-spans-7bc799481a024fc1a6056c9dea5e15d7?d=891dfb3b51df428aa8ce83105132867c&pvs=4#52c9f6bc6d344665b7af4a3d43be0be3

**Review by GuyMoses — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -98,6 +98,7 @@ The `lumigo_opentelemetry` package additionally supports the following configura
   * We support more granular masking using the following parameters. If not given, the above configuration is the fallback: `LUMIGO_SECRET_MASKING_REGEX_HTTP_REQUEST_BODIES`, `LUMIGO_SECRET_MASKING_REGEX_HTTP_REQUEST_HEADERS`, `LUMIGO_SECRET_MASKING_REGEX_HTTP_RESPONSE_BODIES`, `LUMIGO_SECRET_MASKING_REGEX_HTTP_RESPONSE_HEADERS`, `LUMIGO_SECRET_MASKING_REGEX_HTTP_QUERY_PARAMS`, `LUMIGO_SECRET_MASKING_REGEX_ENVIRONMENT`.
 * `LUMIGO_SWITCH_OFF=true`: This option disables the Lumigo OpenTelemetry distro entirely; no instrumentation will be injected, no tracing data will be collected.
 * `LUMIGO_REPORT_DEPENDENCIES=false`: This option disables the built-in dependency reporting to Lumigo SaaS. For more information, refer to the [Automated dependency reporting](#automated-dependency-reporting) section.
+* `LUMIGO_AUTO_FILTER_EMPTY_SQS`: This option enables the automatic filtering of empty SQS messages from being sent to Lumigo SaaS.
```
Comment (GuyMoses) [inline README.md:101]: maybe, explain that it is boolean and defaults to true?

**Review by GuyMoses — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (sagivoululumigo) [reply README.md]: Your right, referenced to the detailed section & added every possible value (true, false & unset) & what it does there

**Pull request merged**